### DISCUSSION TITLE
AttributeError:'tuple' object has no attribute 'rank'

### DISCUSSION THREAD

**Discussion by fernandorannou** [esmf-org/esmf#262] [ESMPy] [answered]
### Requirements

- [X] Reviewed [ESMPy Documentation](https://earthsystemmodeling.org/esmpy/)
- [X] Searched [GitHub Discussions](https://github.com/orgs/esmf-org/discussions?discussions_q=)

### Affiliation(s)

Universidad de Santiago de Chile

### ESMF Version

esmpy

### Issue

Hello
I'm getting the following error when creating a field on a mesh:
Traceback (most recent call last):
  File "/home/rannou/Project/regrid.py", line 205, in <module>
    sphfield = esmpy.Field(sphmesh, name='sphmesh', meshloc=esmpy.MeshLoc.NODE)
  File "/home/rannou/VirtualEnvironments/esmpy/lib/python3.10/site-packages/esmpy/util/decorators.py", line 59, in new_func
    return func(*args, **kwargs)
  File "/home/rannou/VirtualEnvironments/esmpy/lib/python3.10/site-packages/esmpy/api/field.py", line 87, in __init__
    rank = grid.rank
AttributeError: 'tuple' object has no attribute 'rank'

The python lines are:
sphmesh = create_2D_sph_mesh(rcorner, thetacorner)
sphfield = esmpy.Field(sphmesh, name='sphmesh', meshloc=esmpy.MeshLoc.NODE)

As far as I can tell the mesh has been successfully created. The esmpy documentation says:
property rank
Return type
int
Returns
The rank of the Mesh, (i.e. always 1).

I appreciate any suggestions. Thanks

Fer

### Autotag

@billsacks

**Comment by billsacks** [comment, marked as answer]
My first guess is that your `create_2D_sph_mesh` function is not actually returning an ESMPy mesh object but instead is returning a tuple. What do you get if you run this?:

```
type(sphmesh)
```

You should get:

```
esmpy.api.mesh.Mesh
```

You might also want to compare your code and result with the example here:

https://earthsystemmodeling.org/esmpy_doc/release/latest/html/examples.html#create-a-5-element-mesh

**Reply by fernandorannou** [reply]
Thanks for your help. I was returning a mesh plus other objects, but I forgot to list them all on the function call.
Now it works ok.

sphmesh, nodeCoord, nodeOwner, elemType, elemConn, elemCoord = create_2D_sph_mesh(rcorner, thetacorner)

Fer
